pkg/v0/cmd.go

```diff
@@ -160,9 +181,86 @@ func Run(ctx context.Context, logger *logrus.Logger, opts Options) error {
 	return nil
 }
 
+func getTagOrCommit(repo string, opts Options, logger *logrus.Entry) (string, error) {
+	file, err := os.Open(opts.stagingDir + opts.baseRepo + "/go.mod")
+	if err != nil {
+		logger.Fatalf("Unable to open %q: %v", opts.baseRepo, err)
+		return "", err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for i := 1; scanner.Scan(); i++ {
+		str := strings.Trim(scanner.Text(), " \t")
+		if strings.Contains(str, repo) {
+			strs := strings.Split(str, " ")
+			if len(strs) < 2 {
+				return "", fmt.Errorf("Bad format: %q", str)
+			}
+			verstr := strs[1]
+			v, err := semver.NewVersion(verstr)
+			if err != nil {
+				return "", err
+			}
+			// If this does not have a Prerelease, then we just return the version string
+			pre := v.Prerelease()
+			if pre == "" {
+				return verstr, nil
+			}
+			// It's a pre-release version, which we assume is in DATE-COMMIT format
+			pres := strings.Split(pre, "-")
+			if len(pres) != 2 {
+				return "", fmt.Errorf("Bad prerelease: %q", verstr)
+			}
+			// Return the second component, which is a commit SHA
+			return pres[1], nil
+		}
+	}
+	return "", fmt.Errorf("%q not found", repo)
+}
+
+func calculateRepoRefs(ctx context.Context, logger *logrus.Entry, opts Options) (map[string]string, error) {
```

Comment:
Why don't we re-use the logic in the v1 bumper for this?

Reply:
Was there logic? Let me look.

Reply:
The bits that query `go.mod` should be re-usable at least!

Reply:
Yup, I just did it a bit different. Using `go list` ought to be more reliable.